megago/megago.py

```diff
@@ -85,172 +89,15 @@ def read_input(in_file, sep=",", go_sep=";"):
-    
-    Formula of the metric: (2 * info_content(mica) * (1 - freq(mica))) / (info_content(go_id1) + info_content(go_id2))
-    where mica is the most informative common ancestor of go_id1 and go_id2.
-    Metric is implemented according to: Schlicker, A., Domingues, F.S., Rahnenführer, J. et al. A new measure for 
-    functional similarity of gene products based on Gene Ontology. BMC Bioinformatics 7, 302 (2006) 
-    doi:10.1186/1471-2105-7-302 
-    
-
-    Parameters
-    ----------
-    go_id1 : str
-        GO term
-    go_id2 : str
-        GO term
-    go_dag : GODag object
-        GODag object from the goatools package
-    term_counts : dict
-        dictionary: key: GO terms, values: number of occurrences of GO term and its children in body of evidence
+def parse_args():
+    """ Parse command line arguments. This function will exit the program on a command line error!
 
     Returns
     -------
-    float 
-        if go_id1 and go_id2 are from different GO namespaces or either of them misses in the go_dag: NAN_VALUE
-        else: rel metric
-
-    """
-
-    if (go_id1 not in go_dag) or (go_id2 not in go_dag):
-        return NAN_VALUE
-
-    goterm1 = go_dag[go_id1]
-    goterm2 = go_dag[go_id2]
-    if goterm1.namespace == goterm2.namespace:
-        mica_goid = deepest_common_ancestor([go_id1, go_id2], go_dag)
-        freq_mica = get_frequency(mica_goid, term_counts, go_dag)
-        info_content_mica = get_info_content(mica_goid, term_counts, go_dag)
-        info_content1 = get_info_content(go_id1, term_counts, go_dag)
-        info_content2 = get_info_content(go_id2, term_counts, go_dag)
-        # if info content of go_id == 0, although the term occurs multiple times in the body of evidence, the term
-        # counts of the go_id include all term counts in the specific namespace
-        if (info_content1 == 0) and (term_counts.get(go_id1, 0) == 0):
-            info_content1 = get_ic_of_most_informative_ancestor(go_id1, term_counts, go_dag)
-        if (info_content2 == 0) and (term_counts.get(go_id2, 0) == 0):
-            info_content2 = get_ic_of_most_informative_ancestor(go_id2, term_counts, go_dag)
-        if info_content1 + info_content2 == 0:
-            return 0
-        return (2 * info_content_mica * (1 - freq_mica)) / (info_content1 + info_content2)
-    else:    # if goterms are from different GO namespaces (molecular function, cellular component, biological process)
-        return NAN_VALUE
-
-
-def parse_args():
-    """Parse command line arguments.
-    Returns Options object with command line argument values as attributes.
-    Will exit the program on a command line error.
+    Options object with command line argument values as attributes.
     """
-
-    parser = argparse.ArgumentParser(description=__doc__)
+    description = 'Calculate semantic distance for sets of Gene Ontology terms'
+    parser = argparse.ArgumentParser(description=description)
```

Comment:
Why not use the doc string of the module? I introduced the usage of the doc string based on [this](https://realpython.com/documenting-python-code/#documenting-your-python-code-base-using-docstrings)

Reply:
Please don't get me wrong, I am not criticizing your changes but rather would like to know your rationale.

Reply:
I must have made a mistake while resolving a merge conflict. Yours is the right way to go.